### PR TITLE
curl: gate darwin workarounds behind stdenv.isDarwin to make life on linux easier

### LINKED ISSUE #3382: Provide more assistance for users to find SSL certificates (the "unable to get local issuer certificate" problem)
This caused me a big headache yesterday. A solution (for git) is to install cacert and then run GIT_SSL_CAINFO=$HOME/.nix-profile/etc/ca-bundle.crt, but when running outside nixos that environment variable isn't set and there's no warning/indication that you'd need to set anything. 

Basically this means curl/git are broken out of the box.

A nice thing would be to print "Curl has installed but no SSL certificates are currently available, which means all SSL operations will fail. To add SSL certificates, install the cacert package, then set CURL_CA_BUNDLE=$HOME/.nix-profile/etc/ca-bundle.crt in your environment", in bright red, when curl is installed, and a similar message for git.

It would also be nice if the packages just shipped with a certificate list.

### PR DESCRIPTION
**Pull request by SuperSandro2000**
…

Closes #3382

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by FRidh** [comment]
I prefer patching only those that are affected as well, however, I know there are those that have a different view. Their view is to apply it for everyone because that way you can detect easily when a patch does not apply anymore and it will more likely be fixed.

File: pkgs/tools/networking/curl/default.nix [hunk 1]
```diff
@@ -136,6 +131,12 @@ stdenv.mkDerivation rec {
     ++ lib.optionals stdenv.hostPlatform.isWindows [
       "--disable-shared"
       "--enable-static"
+    ] ++ lib.optionals stdenv.isDarwin [
```
Comment (thiagokokada) [inline pkgs/tools/networking/curl/default.nix:134]: Wouldn't it be better to use `stdenv.hostPlatform.isDarwin` instead to take care of the cross-compiling cases here?

**Comment by thiagokokada** [comment]
> I prefer patching only those that are affected as well, however, I know there are those that have a different view. Their view is to apply it for everyone because that way you can detect easily when a patch does not apply anymore and it will more likely be fixed.

Not related to this PR, but I think we need to have a clearer instance about Darwin.

It is clearly a "best effort" platform that causes tons of headaches thanks to the inability of Apple of supplying good sources where we can base our `stdenv` on, that is also constantly changing and breaking things at each update making each upgrade a mess.

Not criticizing the excellent work of the Darwin maintainers, but it really is a platform that is a PITA to work with.

File: pkgs/tools/networking/curl/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/tools/networking/curl/default.nix:134]: isDarwin is an alias to hostPlatform.isDarwin https://github.com/nix-community/home-manager/blob/master/tests/default.nix#L35-L36

**Comment by SuperSandro2000** [comment]
> I prefer patching only those that are affected as well, however, I know there are those that have a different view. Their view is to apply it for everyone because that way you can detect easily when a patch does not apply anymore and it will more likely be fixed.

Normally this is the case but since the flags make the linux experience actually worse I think we can make an exception here.

**Pull request merged**